### PR TITLE
Feat: Support  for Emoji Option in Feed Editor

### PR DESCRIPTION
**Pull request by shrushti2000**
This PR introduces an emoji picker functionality to the feed editor, allowing users to insert emojis into their posts.

https://github.com/user-attachments/assets/ead3dad8-467b-4f18-8149-3dee21b376b9

https://github.com/user-attachments/assets/2c6a3cb3-a633-49d8-8dd2-06c2b6b7f4c1

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/FeedEditor/FeedEditor.tsx [hunk 1]
```diff
@@ -55,6 +57,8 @@ import { FeedEditorProp, MentionSuggestionsItem } from './FeedEditor.interface';
 
 Quill.register('modules/markdownOptions', QuillMarkdown);
 Quill.register(LinkBlot as unknown as Parchment.RegistryDefinition);
+Quill.register('modules/emoji-shortname', ShortNameEmoji, true);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/FeedEditor/FeedEditor.tsx:60]: @shrushti2000 , we can remove this, we just need TextAreaEmoji.

File: openmetadata-ui/src/main/resources/ui/webpack.config.dev.js [hunk 2]
```diff
@@ -116,6 +116,7 @@ module.exports = {
     },
     alias: {
       process: 'process/browser',
+      Quill: path.resolve(__dirname, 'node_modules/quill'),
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/webpack.config.dev.js:119]: @shrushti2000, why do we need this change?

Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/webpack.config.dev.js:119]: Can we please add comments in the code on why we need this?

**Review by chirag-madlani — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/webpack.config.dev.js (hunk 2 above)
Comment (shrushti2000) [reply openmetadata-ui/src/main/resources/ui/webpack.config.dev.js]: Done, added comment for this.

**Review by aniketkatkar97 — APPROVED** [review]

**Pull request merged**